Source/ui_unix/mainwindow.cpp

```diff
@@ -0,0 +1,228 @@
+#include "mainwindow.h"
+#include "settingsdialog.h"
+
+#include "openglwindow.h"
+
+#include <QFileDialog>
+#include <QTimer>
+#include <QWindow>
+
+#include "global.h"
+#include "GSH_OpenGLQt.h"
+#include "tools/PsfPlayer/Source/SH_OpenAL.h"
+
+#include "ui_mainwindow.h"
+
+#define PREFERENCE_UI_SHOWFPS "ui.showfps"
+#define PREFERENCE_UI_SHOWVIRTUALPAD "ui.showvirtualpad"
+#define PREFERENCE_AUDIO_ENABLEOUTPUT "audio.enableoutput"
+#define PREF_CGSH_OPENGL_ENABLEHIGHRESMODE        "renderer.opengl.enablehighresmode"
+#define PREF_CGSH_OPENGL_FORCEBILINEARTEXTURES    "renderer.opengl.forcebilineartextures"
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+
+    openglpanel = new OpenGLWindow;
+
+    QWidget * container = QWidget::createWindowContainer(openglpanel);
+    ui->gridLayout->addWidget(container);
+    connect(openglpanel, SIGNAL(heightChanged(int)), this, SLOT(openGLWindow_resized()));
+    connect(openglpanel, SIGNAL(widthChanged(int)), this, SLOT(openGLWindow_resized()));
+
+    connect(openglpanel, SIGNAL(keyUp(QKeyEvent*)), this, SLOT(keyReleaseEvent(QKeyEvent*)));
+    connect(openglpanel, SIGNAL(keyDown(QKeyEvent*)), this, SLOT(keyPressEvent(QKeyEvent*)));
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+void MainWindow::showEvent(QShowEvent* event)
+{
+    QMainWindow::showEvent(event);
+    initEmu();
+}
+
+void MainWindow::initEmu(){
+    g_virtualMachine = new CPS2VM();
+    g_virtualMachine->Initialize();
+
+    g_virtualMachine->CreateGSHandler(CGSH_OpenGLQt::GetFactoryFunction(openglpanel));
+    SetupSoundHandler();
+
+    g_virtualMachine->CreatePadHandler(CPH_HidUnix::GetFactoryFunction());
+    padhandler = (CPH_HidUnix*)g_virtualMachine->GetPadHandler();
+
+    StatsManager = new CStatsManager();
+    g_virtualMachine->m_ee->m_gs->OnNewFrame.connect(boost::bind(&CStatsManager::OnNewFrame, StatsManager, _1));
+
+    createStatusBar();
+
+    fpstimer = new QTimer(this);
```

Comment:
@jpd002 ya, Qt has a garbage collector ([1](http://silmor.de/qtstuff.garbcol.php), [2](url)) that takes care of this.... I recall reading that on their actual site, but i can't find that link right now.